virttest/libvirt_xml/vm_xml.py

```diff
@@ -613,41 +613,44 @@ def vm_rename(vm, new_name, uuid=None, virsh_instance=base.virsh):
         :param vm: VM class type instance
         :param new_name: new name of vm
         :param uuid: new_vm's uuid, if None libvirt will generate.
-        :return: a new VM instance
+        :return: a new VM instance or raise LibvirtXMLError
         """
+        if vm.name == new_name:
+            logging.warning("New name is same as VM, No need to rename")
+            return vm
+        start_vm = False
         if vm.is_alive():
             vm.destroy(gracefully=True)
-        vmxml = VMXML.new_from_dumpxml(vm_name=vm.name,
-                                       virsh_instance=virsh_instance)
+            start_vm = True
+        vmxml = VMXML.new_from_dumpxml(vm.name, virsh_instance=virsh_instance)
         backup = vmxml.copy()
-        # can't do in-place rename, must operate on XML
+
+        def _cleanup(details=""):
```

Comment:
Yes, Indeed helper function in the middle leads bad influence on reading.
But if we don't define this helper function,
It will need three copy of the helper function when operationes fail.